lora/fuse.py

```diff
@@ -0,0 +1,69 @@
+# Copyright © 2023 Apple Inc.
+
+import argparse
+
+import mlx.core as mx
+import models
+import utils
+from mlx.utils import tree_flatten, tree_unflatten
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="LoRA or QLoRA finetuning.")
+    parser.add_argument(
+        "--model",
+        default="mlx_model",
+        help="The path to the local model directory.",
+    )
+    parser.add_argument(
+        "--save-path",
+        default="lora_fused_model",
+        help="The path to save the fused model.",
+    )
+    parser.add_argument(
+        "--adapter-file",
+        type=str,
+        default="adapters.npz",
+        help="Save/load path for the trained adapter weights.",
+    )
+    parser.add_argument(
+        "--hf-path",
+        help="Path to the original Hugging Face model (required for upload).",
+        type=str,
+        default=None,
+    )
```

Comment:
This was a bit confusing to me, especially when you use a remote base model in `--model`. Perhaps we could allow it to be `None`, default to `--model`, and verify that it's a HF remote repo. We can do that in a followup PR.

Reply:
This is a really good point. I didn't even consider that `--model` could actually be an HF repo. What I could do is check if `model` exists as a path. If it does, then it is a local path that we loaded from and we require `hf-path`. If it doesn't then it must be an HF repo and we can safely set `hf-path` to it.